virtual_ship/sailship.py

```diff
@@ -412,6 +419,47 @@ def _verify_waypoints(
             "Each waypoint should be timed after all previous waypoints"
         )
 
+    # check if all waypoints are in water
+    # this is done by picking an arbitrary provided fieldset and checking if UV is not zero
+
+    # get all available fieldsets
+    available_fieldsets = [
+        fs
+        for fs in [
+            config.adcp_config.fieldset if config.adcp_config is not None else None,
+            config.argo_float_config.fieldset,
+            config.ctd_config.fieldset,
+            config.drifter_config.fieldset,
+            (
+                config.ship_underwater_st_config.fieldset
+                if config.ship_underwater_st_config is not None
+                else None
+            ),
+        ]
+        if fs is not None
+    ]
+    # check if there are any fieldsets, else its an error
+    if len(available_fieldsets) == 0:
+        raise ValueError(
+            "No fieldsets provided to check if waypoints are on land. Assuming no provided fieldsets is an error."
+        )
+    # pick any
+    fieldset = available_fieldsets[0]
+    # get waypoints with 0 UV
+    land_waypoints = [
+        (wp_i, wp)
+        for wp_i, wp in enumerate(waypoints)
+        if fieldset.UV.eval(
+            0, 0, wp.location.lat, wp.location.lon, applyConversion=False
+        )
+        == (0.0, 0.0)
+    ]
```

Comment:
Make this a more general method, so that someone could later change the way in which landpoints are defined (e.g. through a mask-array, instead of zero-velocity condition)

Reply:
see update